File: 0540-single-element-in-a-sorted-array/0540-single-element-in-a-sorted-array.py
class Solution:
    def singleNonDuplicate(self, nums: List[int]) -> int:
        n=len(nums)
        low=0
        high=n-1

        while(low<high):
            guess=(low+high)//2

            if(guess%2==1):
                guess-=1

            if(nums[guess]==nums[guess+1]):
                low=guess+2
            else:
                high=guess

        return nums[low]

File: 0540-single-element-in-a-sorted-array/0540-single-element-in-a-sorted-array.java
class Solution {
    public int singleNonDuplicate(int[] nums) {
        int n=nums.length;
        int low=0;
        int high=n-1;

        while(low<high){
            int guess=(low+high)/2;

            if(guess%2==1){
                guess--;
            }

            if(nums[guess]==nums[guess+1]){
                low=guess+2;
            }
            else{
                high=guess;
            }
        }
        return nums[low];
    }
}